[TypeScript] Allow lab components to have overrides in theme (#21279)
mui-org/material-ui: docs/src/pages/components/about-the-lab/about-the-lab.md

// with yarn
yarn add @material-ui/core
```

## TypeScript

> In order to benefit from the [CSS overrides](/customization/globals/#css) and [default prop customization](/customization/globals/#default-props) with the theme, TypeScript users need to use [module augmentation](/guides/typescript/#customization-of-theme) to extend the default theme structure with the extension components available in the lab:

Author: Maybe mention that we do the module augmentation for them and they just need to include it?

Reviewer: ```suggestion
In order to benefit from the [CSS overrides](/customization/globals/#css) and [default prop customization](/customization/globals/#default-props) with the theme, TypeScript users need to import the following types. Internally, it uses [module augmentation](/guides/typescript/#customization-of-theme) to extend the default theme structure with the extension components available in the lab.
```

```tsx
import type '@material-ui/lab/themeAugmentation';

const theme = createMuiTheme({
  overrides: {

>     MuiThemeline: {

Author: Is this a lab component?

Reviewer 2: ```suggestion
    MuiTimeline: {
```

Reviewer: @CarsonF It will soon be in the lab :)

      root: {

>         backgroundColot: 'red',

Author: ```suggestion
        backgroundColor: 'red',
```
